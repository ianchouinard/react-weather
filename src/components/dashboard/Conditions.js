import React from 'react'
import { store } from './../../store/store'

export const Conditions = (props) => {
  const {conditions} = props
  const unit = store.getState().unitType
  const tempUnit = (unit === 'metric') ? 'C' : 'F'
  const speedUnit = (unit === 'metric') ? 'kph' : 'mph'

  return (
    <div className="conditions">
      <h3>{conditions.weather[0] ? conditions.weather[0].main : 'N/A'}</h3>

      <p>Current Temp</p>
      <h3>{Math.round(conditions.main.temp)}<sup>&deg; {tempUnit}</sup></h3>

      <p>Wind Speed</p>
      <h4>{conditions.wind.speed} <sup>{speedUnit}</sup></h4>
    </div>
  )
}
