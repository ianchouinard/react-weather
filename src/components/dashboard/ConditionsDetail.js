import React from 'react'
import { Units } from '../../classes/Units';

export const ConditionsDetail = (props) => {
  const { main, rain, snow } = props.conditions
  return (
    <div className="condtionsDetail">
      <div className="row">
        <div className="col">
          High Temp: <strong>{ main.temp_max }</strong>
          <sup>&deg; { Units.tempUnit() }</sup>
        </div>
        <div className="col">
          Low Temp: <strong>{ main.temp_min }</strong>
          <sup>&deg; { Units.tempUnit() }</sup>
        </div>
      </div>
      <div className="row">
        <div className="col">
          Humidity: <strong>{ main.humidity }</strong>
          <sup>{ Units.humidityUnit() }</sup>
        </div>
        <div className="col">
          Pressure: <strong>{ main.pressure }</strong>
        </div>
      </div>
      {
        (() => {
          if (snow && (snow['1h'] ||  snow['3h'])) {
            const snow1h = snow['1h'] ? snow['1h'].toFixed(2) : null
            const snow3h = snow['3h'] ? snow['3h'].toFixed(2) : null
            return (
              <div>
                Snow: <strong>{ snow1h || snow3h }</strong>
                <sup>{Units.precipitationUnit()}</sup>
              </div>
            )
          }
        })()
      }

      {
        (() => {
          if (rain && (rain['1h'] ||  rain['3h'])) {
            const rain1h = rain['1h'] ? rain['1h'].toFixed(2) : null
            const rain3h = rain['3h'] ? rain['3h'].toFixed(2) : null
            return (
              <div>
                Rain: <strong>{ rain1h || rain3h }</strong>
                <sup>{ Units.precipitationUnit() }</sup>
              </div>
            )
          }
        })()
      }
    </div>
  )
}
