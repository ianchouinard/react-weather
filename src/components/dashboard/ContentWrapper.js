import React, { Component } from 'react'
import { CurrentConditions } from './current-conditions/CurrentConditions'
import { ForecastScrollContainer } from './future-conditions/ForecastScrollContainer'
import { FutureConditionsChart } from './future-conditions/FutureConditionsChart'

export class ContentWrapper extends Component {

  // Set display of loading bar is data is loading
  showLoadingBar = () => {
    return this.props.isLoadingConditions ? 'block' : 'none'
  }

  render() {
    if (this.props.currentConditionsLoaded) {
      return (
        <div className="dashboardContent">
          <CurrentConditions currentConditions={this.props.currentConditions} />
          <ForecastScrollContainer conditions={this.props.futureConditions} />
          <FutureConditionsChart futureConditions={this.props.futureConditions} />
          <div className="loader" style={{ display:  this.showLoadingBar() }}></div>
        </div>
      )
    }

    return ( 
      <div className="loader"></div>
    )
  }

}
