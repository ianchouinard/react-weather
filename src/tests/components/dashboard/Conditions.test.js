import React from 'react';
import renderer from 'react-test-renderer';
import { Conditions } from './../../../components/dashboard/Conditions'
import { mockCondition } from './../../Mocks'

describe('Conditions', () => {

  const component = renderer.create(
    <Conditions conditions={mockCondition(800)} />
  )

  it('should create the component', () => {
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
