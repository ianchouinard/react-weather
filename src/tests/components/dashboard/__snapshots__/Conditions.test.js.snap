// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Conditions should create the component 1`] = `
<div
  className="conditions"
>
  <h3 />
  <p>
    Current Temp
  </p>
  <h3>
    32
    <sup>
      Â° 
      F
    </sup>
  </h3>
  <p>
    Wind Speed
  </p>
  <h4>
    0
     
    <sup>
      mph
    </sup>
  </h4>
</div>
`;
